### DISCUSSION TITLE
Problem with creating a custom linear gradient effect.

### DISCUSSION THREAD

**Discussion by BearCooder** [tailwindlabs/tailwindcss#11243] [Help] [answered]
Hey,

I want to add into my tailwind config [this](https://codepen.io/joshuapekera/pen/xGjMMq) shiny text effect.

When reduced into the necessary CSS part its this:

```
.shimmer {
    text-align: center;
    color: rgba(255, 255, 255, 0.1);
    background: linear-gradient(to right, black, white);
    background-size: 125px 100%;
    background-clip: text;
    animation-name: shimmer;
    animation-duration: 2s;
    animation-iteration-count: infinite;
    background-repeat: no-repeat;
    background-position: 0 0;
    background-color: #222;
}


@keyframes shimmer {
    0% {
        background-position: top left;
    }
    100% {
        background-position: top right;
    }
}

```

So what I did is I acded into my tailwind config the following code:

```
  theme: {
    extend: {
      keyframes: {
        "shimmer": {
          "0%": { backgroundPosition: "top left"},
          "100%": { backgroundPosition: "top right"},
        }
      },
      animation: {
        shimmer: "2s infinite text-center bg-gradient-to-r from-black to-white bg-black"
      },
    },
  },
  
```
  
 So I added the keyframes and then in my animation I use it together with the linear gradient. This should be enough to be able to use the custom animation when I write "animate-shimmer" in my text? However I guess I overlooked something. Can you please help me?

**Comment by wongjn** [comment, marked as answer]
You have too many custom identifier strings (`text-center`, `bg-gradient-to-r`, `from-black`, `to-white` and `bg-black`) in your `animation` value, and none of these correspond to the `@keyframe` you declared. you should use the same identifier, `shimmer`:

```diff
 animation: {
-  shimmer: "2s infinite text-center bg-gradient-to-r from-black to-white bg-black"
+  shimmer: "2s infinite shimmer"
 },
```

Apply the other properties onto the element along with `animate-shimmmer`:

```html
<div class="animate-shimmer bg-gradient-to-r from-black via-white to-black bg-black text-5xl bg-clip-text text-transparent text-center bg-[length:125px_100%] bg-no-repeat w-max pr-[120px]">
  Foo Bar
</div>
```

[Live example on Tailwind Play](https://play.tailwindcss.com/m8ftVuQnS5)

**Reply by BearCooder** [reply]
Thank you very much! Now its clear to me
